src/providers/trusted_service/key_management.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::Provider;
+use crate::authenticators::ApplicationName;
+use crate::key_info_managers::KeyTriple;
+use crate::key_info_managers::{self, ManageKeyInfo};
+use crate::providers::mbed_crypto::key_management::{
+    create_key_id, get_key_id, insert_key_id, remove_key_id,
+};
+use log::error;
+use parsec_interface::operations::{
+    psa_destroy_key, psa_export_public_key, psa_generate_key, psa_import_key,
+};
+use parsec_interface::requests::{ProviderID, ResponseStatus, Result};
+use parsec_interface::secrecy::ExposeSecret;
+
+/// Check whether any key info exists for a given key triple in the Key Info Manager
+pub fn key_info_exists(key_triple: &KeyTriple, store_handle: &dyn ManageKeyInfo) -> Result<bool> {
+    store_handle
+        .exists(key_triple)
+        .map_err(key_info_managers::to_response_status)
+}
```

Comment:
This can be replaced with the implemented on `ManageKeyInfo` `does_not_exist` if you wished, to also be able to directly do `store_handle.does_not_exist()?;`

Reply:
Yeah, good shout, forgot that exists now